plugins/by-name/fidget/default.nix

```
@@ -33,651 +27,568 @@ let
         Default annotation for error items.
       '';
 
-      priority = helpers.defaultNullOpts.mkUnsignedInt 50 ''
+      priority = defaultNullOpts.mkUnsignedInt 50 ''
         Order in which group should be displayed.
       '';
 
-      skipHistory = helpers.defaultNullOpts.mkBool true ''
+      skip_history = defaultNullOpts.mkBool true ''
         Whether progress notifications should be omitted from history.
       '';
+
+      update_hook = defaultNullOpts.mkNullable (with types; either rawLua (enum [ false ])) false ''
+        Called when an item is updated
+      '';
     };
   };
 in
-{
-  # TODO: deprecation warnings introduced 2023/12/22: remove in early February 2024
-  imports =
-    map
-      (
-        oldOption:
-        mkRemovedOptionModule
-          [
-            "plugins"
-            "fidget"
-            oldOption
-          ]
-          ''
-            Nixvim: The fidget.nvim plugin has been completely rewritten. Hence, the options have changed.
-            Please, take a look at the updated documentation and adapt your configuration accordingly.
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "fidget";
+  packPathName = "fidget.nvim";
+  package = "fidget-nvim";
+
+  maintainers = [ lib.maintainers.HeitorAugustoLN ];
 
-            > https://github.com/j-hui/fidget.nvim
+  settingsOptions = {
+    progress = {
+      poll_rate =
+        defaultNullOpts.mkNullableWithRaw (with types; either ints.unsigned (enum [ false ])) 0
           ''
-      )
-      [
-        "text"
-        "align"
-        "timer"
-        "window"
-        "fmt"
-        "sources"
-        "debug"
-      ];
+            How and when to poll for progress messages.
+
+            Set to `false` to disable polling altogether. You can still manually pool
+            progress messages by calling |fidget.progress.poll|.
+          '';
 
-  options = {
-    plugins.fidget = lib.nixvim.neovim-plugin.extraOptionsOptions // {
-      enable = mkEnableOption "fidget-nvim";
+      suppress_on_insert = defaultNullOpts.mkBool false ''
+        Suppress new messages while in insert mode.
+      '';
 
-      package = lib.mkPackageOption pkgs "fidget-nvim" {
-        default = [
-          "vimPlugins"
-          "fidget-nvim"
-        ];
-      };
+      ignore_done_already = defaultNullOpts.mkBool false ''
+        Ignore new tasks that are already complete.
+      '';
+
+      ignore_empty_message = defaultNullOpts.mkBool false ''
+        Ignore new tasks that don't contain a message.
+      '';
+
+      clear_on_detach =
+        defaultNullOpts.mkNullable
+          (
+            with types;
+            oneOf [
+              strLuaFn
+              rawLua
+              (enum [ false ])
+            ]
+          )
+          ''
+            function(client_id)
+              local client = vim.lsp.get_client_by_id(client_id)
+              return client and client.name or nil
+            end
+          ''
```

```suggestion
          (literalLua ''
            function(client_id)
              local client = vim.lsp.get_client_by_id(client_id)
              return client and client.name or nil
            end
          '')
```

`literalLua` needs inheriting from `lib.nixvim`.

Without this, the default will show as a "string", however we want it to show as a `mkRaw "string"`.